design/overview.md

```diff
@@ -73,4 +78,14 @@ Note: X% is some security factor for bootstrapping. 67% is sufficient to release
 
 ## Relaying Cosmos to ETH
 
-**TODO**
\ No newline at end of file
+* Someone (permissionlessly) sends MsgSendToEth this creates an entry in the 'peggy bridge tx pool' this pool is in the Cosmos chain store and contains all transactions not yet bridged. Funds of the specified demon are then burned from the users account if that demon is valid to bridge, otherwise the tx is rejected. 
+* Someone (permissionlessly) sends a MsgRequestBatch, this produces a new `Transaction batch` in the `Peggy Bridge Tx pool`. The creation of this batch occurs in ComsosSDK and is entierly deterministic, ordering transactions from highest to lowest fee and including them in the `Transaction batch` until there are no more transactions or the batch size limit is reached.
+* The `Orchestrator Set` observes the `Peggy Bridge Tx pool` via REST endpoints. When a new batch appears they sign the batch and submit a MsgConfirmBatch containing their Ethereum signature over the batch.
+* Once enough signatures are collected (66% of voting power) any `Orcestrator` can collect these signatures from the REST endpoint, assemble an Ethereum transaction and submit the batch.  
+* The `Orchestrator Set` observes the Ethereum blockchain via their trusted Ethereum node. Once the batch has entered the blockchain and `EthBlockDelay` has elapsed an `Orchestrator` will send MsgBatchInChain. Containing a signed attestation that the block has entered the Ethereum chain, once 66% of all voting power has confirmed the batch the batch is marked as complete. The batch is removed from the `Peggy Bridge Tx pool` and the MsgSendToEth flow is now fully complete.
```

Comment:
The **Orchestrators** observe the Ethereum blockchain via their trusted Ethereum node. Once the batch has entered the blockchain and `EthBlockDelay` has elapsed **every** `Orchestrator` will send MsgBatchInChain. Containing a signed attestation that the block has entered the Ethereum chain, once 66% of all voting power in the **Orchestrator Set** has confirmed the batch, the batch is marked as complete.

Reply:
Might be cleaner to think of this in two parts:

- A general Ethereum event feed which incorporates a subset of Ethereum's events into the Cosmos state machine state
- The actual logic around when to remove a batch from the pool

Reply:
I want to highlight that we have similar issues in calculating the 66% threshold as mentioned above with an update in the **Orchestrator Set**. It is a nitty detail but if we could have this defined here the implementation and doc would be clear.

Reply:
If we can make proofs of ethereum state on the cosmos state machine, that makes a lot of things cleaner and more secure 

However, afaik there is no way to do spv on ethereum. We just need to prove that certain tx were included in a valid ethereum block and were successful, and not prove ethereum state, which should be easier. But I don't know if that is feasible.

@jtremback as the resident ethereum expert, could you explain or link to some docs on how we could prove this on cosmos state machine. I think it would greatly improve the process

Reply:
This is the best I found, but still underdevelopment...

https://eth.wiki/en/concepts/light-client-protocol

Reply:
This is interesting descripción of eos-eth bridge using the header verification

https://blog.kyber.network/waterloo-a-decentralized-practical-bridge-between-eos-and-ethereum-c25b1698f010

Would it make sense to do something similar in cosmos? We could use a lot of existing go code for the ethereum side, and validation shouldn't be expensive for a native module

Reply:
We are not getting into validating Ethereum block state on the chain except for very specific slashing conditions.

That way lies madness and a 2 year development process. Simply having validators attest will have to do. That being said like @jtremback  said we can have them attest to any Ethereum state we need. But I don't think we actually need much. 

> I want to highlight that we have similar issues in calculating the 66% threshold as mentioned above with an update in the **Orchestrator Set**. It is a nitty detail but if we could have this defined here the implementation and doc would be clear.

I'm not quite sure I follow. Are you saying there are problems where the Orchestrator set is changing while the signatures are being made? 

I'll add a new definition of 'confirmed'

Reply:
@ethanfrey the validators all run an Ethereum heavy client (geth or whatever). For that reason there is no need for light client stuff.

Reply:
> We are not getting into validating Ethereum block state on the chain except for very specific slashing conditions.

> That way lies madness and a 2 year development process. Simply having validators attest will have to do

That is good. I think I will add a section defining what our "light clients" are. Our Cosmos "light client" on Ethereum is the MutliSig Set witnessing a transaction. Our Ethereum "light client" on Cosmos is the Orchestrator Set witnessing a transaction.

We don't need formal light clients, as each of these "light clients" we invent here have the same security set as the validator set of the Cosmos Chain, which is already a maximum security limit.